projects/gsc/experiments/gsc_vdrop_pruning.py

```diff
@@ -0,0 +1,513 @@
+
+        # Optimizer args
+        num_pruning_iterations = assignments.get("num_pruning_iterations", 5)
+        epochs_per_pruning_cycle = assignments.get("epochs_per_pruning_cycle", 4)
+        max_lr = assignments.get("max_lr", 1.5)
+        reg_scalar_max_value = assignments.get("reg_scalar_max_value", 0.1)
+        reg_scalar_final_value = assignments.get("reg_scalar_final_value", 0.001)
+        momentum_max_value = assignments.get("momentum_max_value", 0.75)
+
+        cycle_lr_args = dict(
+            max_lr=max_lr,
+            pct_start=0.0625,
+            anneal_strategy="linear",
+            base_momentum=0.6,
+            max_momentum=momentum_max_value,
+            cycle_momentum=True,
+            final_div_factor=1000.0,
+        )
+        config["multi_cycle_lr_args"] = (
+            (
+                (
+                    0,
+                    dict(
+                        max_lr=1.0,
+                        pct_start=0.2,
+                        anneal_strategy="linear",
+                        base_momentum=0.6,
+                        max_momentum=0.75,
+                        cycle_momentum=True,
+                        div_factor=6.0,
+                        final_div_factor=1000.0,
+                    ),
+                ),
+            )
+            + tuple(
+                [
+                    (30 + prune_iteration * epochs_per_pruning_cycle, cycle_lr_args)
+                    for prune_iteration in range(num_pruning_iterations)
+                ]
+            )
+            + tuple(
+                (30 + num_pruning_iterations * epochs_per_pruning_cycle, cycle_lr_args)
+            )
+        )
+        config["reg_scalar"] = make_reg_schedule(
+            epochs=NUM_EPOCHS,
+            pct_ramp_start=10 / num_epochs,
+            pct_ramp_end=30 / num_epochs,
+            peak_value=reg_scalar_max_value,
+            pct_drop=(30 + num_pruning_iterations * epochs_per_pruning_cycle)
+            / num_epochs,
+            final_value=reg_scalar_final_value,
+        )
+
+        config["prune_schedule"] = tuple(
+            [
+                (
+                    30 + prune_iteration * epochs_per_pruning_cycle,
+                    prune_iteration / num_pruning_iterations,
+                )
+                for prune_iteration in range(num_pruning_iterations)
+            ]
+        )
+
+    @classmethod
+    def get_execution_order(cls):
+        eo = super().get_execution_order()
+        eo["update_config_with_suggestion"].append(
+            "SNRPruningParams.update_config_with_suggestion"
+        )
+        return eo
+
+
+# Initial ~30 epochs one large cycle, followed by pruning iterations
+# how many pruning iterations, and how long is each iteration (num epochs/pruning cycle)
+# how should the learning rate and momentum change? (how high LR should go)
+# how high should the regularization warmup go?
+
+SIGOPT_GSC_VDROP_SNR_PRUNING = deepcopy(GSC_VDROP_SNR_PRUNING)
```

Comment:
We should document the best hyperparameters somewhere, maybe in a dedicated non-SigOpt config that we can run locally.